src/audio/audio_engine.cpp

```diff
@@ -0,0 +1,50 @@
+#include "audio_engine.hpp"
+
+namespace AbaddonClient::Audio {
+
+AudioEngine::AudioEngine(Context &context) noexcept :
+    m_context(context) {}
+
+AudioEngine::~AudioEngine() noexcept {
+    StopTimer();
+}
+
+bool AudioEngine::PlaySound(std::string_view file_path) noexcept {
+    StopTimer();
+
+    const auto result = m_engine.LockScope([this, &file_path](std::optional<Miniaudio::MaEngine> &engine) {
+        if (!engine) {
+            engine = Miniaudio::MaEngine::Create(m_context.GetEngineConfig());
+            if (!engine) {
+                return false;
+            }
+        }
+
+        return engine->PlaySound(file_path);
+    });
+
+    StartTimer();
+    return result;
+}
+
+void AudioEngine::StartTimer() noexcept {
+    // NOTE: I am not using g_timeout_add_seconds_once here since we want to own the source and destroy it manually
+    // g_source_remove throws an error on destroyed source
+    m_timer_source = g_timeout_source_new_seconds(5);
```

Comment:
im guessing this is meant to kill the engine if no sounds are played for 5 seconds? might it just be simpler to keep it around anyways? or maybe we can just call `ma_engine_stop` to avoid having to create it over and over.
also im pretty sure there are glibmm bindings for this but thats not a big deal

Reply:
I was trying to match the behavior of the official client, it spawns source as needed with the timeout of 5 seconds.  `ma_engine_stop` seems to just stop the device's callbacks internally, it still keeps the source intact and connected.

glibmm has `SignalTimeout::connect_seconds_once`, but it doesn't seem to provide a way to destroy/reset the timeout.